store/versioned_cachingstore.go

```diff
@@ -9,14 +9,138 @@ import (
 )
 
 var (
 	separator = "|"
+	//Prometheus metrics
+	getDuration    metrics.Histogram
+	hasDuration    metrics.Histogram
+	deleteDuration metrics.Histogram
+	setDuration    metrics.Histogram
+
+	cacheHits   metrics.Counter
+	cacheErrors metrics.Counter
+	cacheMisses metrics.Counter
 )
 
+type CachingStoreLogger struct {
+	logger *loom.Logger
+}
+
+func (c CachingStoreLogger) Printf(format string, v ...interface{}) {
+	c.logger.Info(format, v)
+}
+
+func init() {
+	const namespace = "loomchain"
+	const subsystem = "versioned_caching_store"
+
+	getDuration = kitprometheus.NewSummaryFrom(
+		stdprometheus.SummaryOpts{
+			Namespace:  namespace,
+			Subsystem:  subsystem,
+			Name:       "get",
+			Help:       "How long Versioned CachingStore.Get() took to execute (in miliseconds)",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		}, []string{"error", "isCacheHit"})
+
+	hasDuration = kitprometheus.NewSummaryFrom(
+		stdprometheus.SummaryOpts{
+			Namespace:  namespace,
+			Subsystem:  subsystem,
+			Name:       "has",
+			Help:       "How long Versioned CachingStore.Has() took to execute (in miliseconds)",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		}, []string{"error", "isCacheHit"})
+
+	deleteDuration = kitprometheus.NewSummaryFrom(
+		stdprometheus.SummaryOpts{
+			Namespace:  namespace,
+			Subsystem:  subsystem,
+			Name:       "delete",
+			Help:       "How long Versioned CachingStore.Delete() took to execute (in miliseconds)",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		}, []string{"error"})
+
+	setDuration = kitprometheus.NewSummaryFrom(
+		stdprometheus.SummaryOpts{
+			Namespace:  namespace,
+			Subsystem:  subsystem,
+			Name:       "set",
+			Help:       "How long Versioned CachingStore.Set() took to execute (in miliseconds)",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		}, []string{"error"})
+
+	cacheHits = kitprometheus.NewCounterFrom(
+		stdprometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "cache_hit",
+			Help:      "Number of cache hit for get/has",
+		}, []string{"store_operation"})
+
+	cacheMisses = kitprometheus.NewCounterFrom(
+		stdprometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "cache_miss",
+			Help:      "Number of cache miss for get/has",
+		}, []string{"store_operation"})
+
+	cacheErrors = kitprometheus.NewCounterFrom(
+		stdprometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "cache_error",
+			Help:      "number of errors encountered while doing any operation on cache",
+		}, []string{"cache_operation"})
+
+}
+
+type CachingStoreConfig struct {
+	CachingEnabled bool
+	// CachingEnabled may be ignored in some configurations, this will force enable the caching
+	// store in those cases.
+	// WARNING: This should only used for debugging.
+	DebugForceEnable bool
```

Comment:
This configuration is not used anymore so we can remove it.

Reply:
Done.